NitroxClient/GameLogic/Spawning/WorldEntities/ReefbackWorldEntitySpawner.cs

```diff
@@ -25,15 +25,15 @@ public IEnumerator SpawnAsync(WorldEntity entity, Optional<GameObject> parent, E
                 result.Set(Optional.Empty);
                 yield break;
             }
-            ReefbackLife life = reefback.Value.GetComponent<ReefbackLife>();
-            if (life == null)
+
+            if (!reefback.Value.TryGetComponent(out ReefbackLife life))
             {
                 result.Set(Optional.Empty);
                 yield break;
             }
 
             life.initialized = true;
-            life.SpawnPlants();
+            yield return life.SpawnPlants();
```

Comment:
Why should this causing duplicate entities? Because of the `CoroutineUtils.waitForNextFrame`?

Reply:
Because the code right below it loops through child entities and spawns them.  This might include plants, and, if so, you would have two plants in the world (one with NitroxEntity and the other without).